hideContentUntilPendingUpdate should not unconditionally create a process throttler activity.

mattwoodrow opened the pull request:
#### 977b091711224b2a2d8dffaaf449d13d064d9808
<pre>
hideContentUntilPendingUpdate should not unconditionally create a process throttler activity.
<a href="https://bugs.webkit.org/show_bug.cgi?id=272423">https://bugs.webkit.org/show_bug.cgi?id=272423</a>
&lt;<a href="https://rdar.apple.com/122365213">rdar://122365213</a>&gt;

Reviewed by Tim Horton.

hideContentUntilPendingUpdate uses a round-trip to the WebContent process, where the reply is
bundled into the following layer transaction. This ensures that any previous IPC messages
from the UI process have been received, and we remain hidden until the next layer transaction
sent after those messages.

We don&apos;t need to hold a background activity and keep the WebProcess unsuspended during this
process, and applicationDidFinishSnapshottingAfterEnteringBackground is explicitly trying
to suspend with the intent that the un-hiding happens on the first paint after returning
from the background.

* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::hideContentUntilPendingUpdate):

Canonical link: <a href="https://commits.webkit.org/277398@main">https://commits.webkit.org/277398@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b82997c668e03eb054027a2ad8549ba622118a87

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47378 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/26558 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50032 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50061 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43426 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32076 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24018 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38573 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/47959 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24088 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/40822 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/19893 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/21526 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5421 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/43718 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/42396 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/51937 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22408 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/18753 "Found 1 new test failure: media/modern-media-controls/overflow-support/dropped-when-narrow.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/45870 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/23683 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/44901 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10479 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/24469 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23402 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/ios/ViewGestureControllerIOS.mm:
```diff
@@ -376,7 +376,7 @@ - (BOOL)isNavigationSwipeGestureRecognizer:(UIGestureRecognizer *)recognizer
                 gestureController->willCommitPostSwipeTransitionLayerTree(true);
         });
-        drawingArea->hideContentUntilPendingUpdate();
+        drawingArea->hideContentUntilPendingUpdate(WebProcessProxy::ShouldStartProcessThrottlerActivity::Yes);
```
I think you could get away with this being a No as well. It's the status quo before this series of patches, it's only relevant if you bg the app after swiping, and presumably would just hold the process awake needlessly if you were to do so.

mattwoodrow committed:
hideContentUntilPendingUpdate should not unconditionally create a process throttler activity.
https://bugs.webkit.org/show_bug.cgi?id=272423
<rdar://122365213>

Reviewed by Tim Horton.

hideContentUntilPendingUpdate uses a round-trip to the WebContent process, where the reply is
bundled into the following layer transaction. This ensures that any previous IPC messages
from the UI process have been received, and we remain hidden until the next layer transaction
sent after those messages.

We don't need to hold a background activity and keep the WebProcess unsuspended during this
process, and applicationDidFinishSnapshottingAfterEnteringBackground is explicitly trying
to suspend with the intent that the un-hiding happens on the first paint after returning
from the background.

* Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm:
(WebKit::RemoteLayerTreeDrawingAreaProxy::hideContentUntilPendingUpdate):

Canonical link: https://commits.webkit.org/277398@main
